### PR TITLE
Added missing lapack docs

### PR DESCRIPTION
**Pull request by kshyatt**
Some of the factorization methods needed documenting.

### CODE REVIEW

File: base/linalg/lapack.jl [hunk 1]
```diff
@@ -594,17 +594,86 @@ code which indicates success (`info = 0`), a singular value in `U`
...
+Returns `A`, modified in-place, and `T`, which contains upper triangular block
+reflectors which parameterize the elementary reflectors of the factorization.
+"""
 geqrt3!{T<:BlasFloat}(A::StridedMatrix{T}) = (n=size(A,2); geqrt3!(A,similar(A,T,n,n)))
+"""
+    geqrf!(A) -> (A, tau)
+
+Compute the `QR` factorization of `A`, `A = QR`.
+
+Returns `A`modified in-place, and `tau`, which contains scalars
```
Comment (tkelman) [inline base/linalg/lapack.jl]: space after `A`

Comment (kshyatt) [inline base/linalg/lapack.jl]: oops

File: base/linalg/lapack.jl [hunk 2]
```diff
@@ -594,17 +594,86 @@ code which indicates success (`info = 0`), a singular value in `U`
...
+which parameterize the elementary reflectors of the factorization.
+"""
 gerqf!{T<:BlasFloat}(A::StridedMatrix{T}) = ((m,n)=size(A); gerqf!(A,similar(A,T,min(m,n))))
 
+"""
+    geqp3!(A, jpvt) -> (A, jpvt, tau)
+
+Compute the pivoted `QR` factorization of `A`, `AP = QR` using BLAS level 3.
+`P` is a pivoting matrix, represented by `jpvt`. `jpvt` must have length length
```
Comment (tkelman) [inline base/linalg/lapack.jl]: duplicate length

**Comment by ViralBShah** [comment]
These docs are great to have. The only one worry I have is that this is not a user facing API we want to commit to support/deprecate etc., even though we will most likely follow that procedure.

Should we add a disclaimer in the lapack docs section along these lines?

**Comment by andreasnoack** [comment]
Yes. I think it would be good point that out.

**Comment by kshyatt** [comment]
I added the bit about support. Does this look ok to everyone?

**Comment by andreasnoack** [comment]
Yes.

**Pull request merged**